src/net/udp.rs

```diff
@@ -0,0 +1,77 @@
+use std::io::Result as IoResult;
+use std::mem::ManuallyDrop;
+use std::os::unix::io::AsRawFd;
+
+use tokio::net::ToSocketAddrs;
+
+use crate::buf::{IoBuf, IoBufMut};
+use crate::driver::{Op, SharedFd};
+
+/// An `io_uring`-backed implementation of the User Datagram Protocol
+pub struct UdpSocket {
+    fd: SharedFd,
+    _inner_sock: ManuallyDrop<std::net::UdpSocket>,
+}
+
+impl UdpSocket {
```

Comment:
There should also be an explicit `close` fn, similar to the `File` type.

Reply:
There are a lot of functions this needs added, I was planning to add a bunch like close, send_to, etc, in another PR.